packages/onboarding/src/step-container-v2/components/StepContainerV2/StepContainerV2.tsx

```diff
@@ -50,15 +50,51 @@ export const StepContainerV2 = ( {
 		[ isMediumViewport, isLargeViewport ]
 	);
 
+	const topBarRef = useRef< HTMLDivElement >( null );
+	const [ topBarHeight, setTopBarHeight ] = useState( 0 );
+
+	const stickyBottomBarRef = useRef< HTMLDivElement >( null );
+	const [ stickyBottomBarHeight, setStickyBottomBarHeight ] = useState( 0 );
+
+	useLayoutEffect( () => {
+		const observer = new ResizeObserver( ( entries ) => {
+			entries.forEach( ( entry ) => {
+				const element = entry.target;
+
+				if ( element === topBarRef.current ) {
+					setTopBarHeight( entry.contentRect.height );
+				} else if ( element === stickyBottomBarRef.current ) {
+					setStickyBottomBarHeight( entry.contentRect.height );
+				}
+			} );
+		} );
+
+		if ( topBarRef.current ) {
+			observer.observe( topBarRef.current );
+		}
+		if ( stickyBottomBarRef.current ) {
+			observer.observe( stickyBottomBarRef.current );
+		}
+
+		return () => {
+			observer.disconnect();
+		};
+	}, [] );
+
 	return (
 		<StepContainerV2InternalProvider value={ stepContainerContextValue }>
 			<div
 				className={ clsx( 'step-container-v2', {
 					'medium-viewport': isMediumViewport,
 					'large-viewport': isLargeViewport,
 				} ) }
+				style={ {
+					// @ts-expect-error -- These are valid CSS variables.
```

Comment:
I prefer casting the object with ` as React.CSSProperties`. It's more explicit about what exactly you want TS to ignore. Currently it will ignore typos in the `topBarHeight` variable name.